libs/pricing_model/src/lib.rs

```diff
@@ -0,0 +1,338 @@
+use std::borrow::Cow;
+use std::collections::HashMap;
+
+/// Maintains latest [`Segment`] for each of the branches, faciliates the pricing model
+/// calculations.
+pub struct Storage<K: 'static> {
+    segments: Vec<Segment>,
+    branches: HashMap<K, usize>,
```

Comment:
What are the key and value here?

Comparing with the original hacky version of this that I wrote, the key was a branch name, as a String. Perhaps it should be TimelineId here?

And the value is an index into the `segments` array. Needs a comment

Reply:
Extracted `K` to support strings in the original scenarios, `Option<TimelineId>` when used from pageserver.

I didn't want to use TimelineId not to change the existing scenarios. Key being generic comes at the cost of a bit of boilerplate, but it seems quite minimal.